src/target/riscv/riscv-013_reg.c

```diff
@@ -92,26 +93,165 @@ static const struct reg_arch_type *riscv013_gdb_regno_reg_type(uint32_t regno)
 	return &riscv011_reg_type;
 }
 
-static int riscv013_init_reg(struct target *target, uint32_t regno)
+static int init_cache_entry(struct target *target, uint32_t regno)
 {
-	return riscv_reg_impl_init_one(target, regno, riscv013_gdb_regno_reg_type(regno));
+	return riscv_reg_impl_init_cache_entry(target, regno, riscv013_gdb_regno_reg_type(regno));
 }
 
-int riscv013_reg_init_all(struct target *target)
+static int examine_xlen(struct target *target)
```

Comment:
I wonder if `riscv013_reg_examine_all()` and its companion `examine_*` functions should rather go to `riscv-013.c`.

The reason is that `riscv-013_reg` module provides the register access (lower-level functions) but the examine logic is the user of it (higher-level algorithms). So to me they belong to a higher level module -- `riscv-013.c` or in future maybe `riscv-013_examine.c`.

@en-sc Please let me know what you think about it. Thanks.

Reply:
IMHO, `riscv013_reg_examine_all()` deals with initialization of the register cache (essentially a constructor of RISC-V register cache). It needs to know much more about the inner workings of register access on RISC-V then any other interface function. So I'd like to leave `riscv013_reg_examine_all()` in `riscv-013_reg.c`.

Reply:
Ok, let's leave it as is.